### DISCUSSION TITLE
Inaccessible knowledge base and internal server errors, potentially due to file removal

### DISCUSSION THREAD

**Discussion by JIW-roosi** [open-webui/open-webui#6354] [Issues]
# Bug Report

## Installation Method
Open WebUI Helm Charts and Image

## Environment

- **Open WebUI Version:** v0.3.32

- **Operating System:**  azure kubernetes service (Ubuntu 22.4.)

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
Initialised chat with a bot which has access to a knowledge base which contains two files. Expected to receive answers based on the knowledge base. 

## Actual Behavior:
Same bot had been tested a couple of days before and now failed to produce answers after the pod had been restarted.   

The container logs reveal, that during the initial chat the knowledge base could not be found (see attached log) and subsequent chats with the same bot only returned a much shorter error, stating that the collection was of a NoneType. 

The knowledge base, however, can be found using the API and appears to be properly configured and linked to in the UI. 
The body returned by the GET /knowledge/ endpoint contains three file ids, while only two are listed in the UI, one having previously been removed (through the UI knowledge workspace). 
Attempts to list all files through the API (endpoint GET /files/) result in an internal server error, while files can be retrieved using the /files/{id} endpoint, except the one surplus file which returns a 404 error. 

## Description

**Bug Summary:**
id of a deleted file remains in the knowledge base, even though the file itself appears to have been properly removed from the backend. 
This appears to be causing internal server errors and interfere with access of the knowledge base. 

## Reproduction Details

**Steps to Reproduce:**
- create knowledge base and upload several files
- remove one of the files form the knowledge base
- create model and give it access to the knowledge base
- restart pod
- chat with the model

## Logs and Screenshots

**Docker Container Logs:**
[Initial chat knowledge base error logs](https://github.com/user-attachments/files/17392859/initial_chat_knowledge_base_error.txt)
[Subsequent chat knowledge base error logs](https://github.com/user-attachments/files/17392876/subsequent_chat_kb_error.txt)
[GET /files/ error logs](https://github.com/user-attachments/files/17392888/error.api.files.call.txt)

**Screenshots/Screen Recordings (if applicable):**
<img width="1127" alt="kb_by_id" src="https://github.com/user-attachments/assets/49c10052-983c-4269-ba3d-a5ff1c6cf32a">
<img width="1063" alt="get_files" src="https://github.com/user-attachments/assets/ecb8817e-5f6b-467f-a7f3-d7c2ac5fd143">

**Comment by tjbck** [comment]
I'm unable to reproduce the issue on my end, I'd greatly appreciate if you could provide us with the files you used and the exact steps you took in thorough detail!

**Reply by JIW-roosi** [reply]
I'm afraid I may not be at liberty to share the files themselves. They are pdf files containing numbered lists.

The initial file contained a list of planned features and a list of benefits of joint development. It was later deleted and replaced with two separate pdfs each containing one of the list. 
In either case the files seem to have been correctly parsed, as the text preview within the knowledge ui section was and is correct.
Chats with the bot having access to them also had worked as expected.  

A couple of days after the initial tests the pod on which owui is running had been restarted. Queries after that only resulted in a reply that no information could be found after the knowledge base had been searched.

I hope that this will be helpful.

**Reply by JIW-roosi** [reply]
Small update: 

Today a similar issue arose in another instance/pod running owui. 
Importing a json model file through the ui information about the knowledge base was imported as well. 
The files had, at the time, not been uploaded or collated into said knowledge base. 
As the files where uploaded, the knowledge base received a different name, wherefore the automatically imported one was removed and the new one (containing actually uploaded files) was attached to the model. 
Calls to the API endpoint GET files now results in an internal server error, while files can be looked for by id, the API returning both information and content. 

It appears as though the errors may be caused by the deletions of files and remainder of traces thereof.

**Reply by tjbck** [reply]
Hmmm, I'm still a bit puzzled as I'm unable to reproduce the issue on my end. I did make some adjustments based on your description of the issue, so let me know if 0.3.34 fixes your issue!

**Reply by JIW-roosi** [reply]
We're puzzled as well. Updating did not help, unfortunately (we've just tested 0.3.35)
